src/controllers/queryRunner.ts

```diff
@@ -222,6 +223,29 @@ export default class QueryRunner {
         this.eventEmitter.emit('batchComplete', batch);
     }
 
+    // Refreshes the webview panel with the query results when tabs are changed
+    public async refreshQueryTab(): Promise<boolean> {
+        for (let batchId = 0; batchId < this.batchSets.length; batchId++) {
+            const batchSet = this.batchSets[batchId];
+            this.eventEmitter.emit('batchStart', batchSet);
+            let executionTime = <number>(Utils.parseTimeString(batchSet.executionElapsed) || 0);
+            this._totalElapsedMilliseconds += executionTime;
+            if (executionTime > 0) {
+                // send a time message in the format used for query complete
+                this.sendBatchTimeMessage(batchSet.id, Utils.parseNumAsTimeString(executionTime));
+            }
+            this.eventEmitter.emit('batchComplete', batchSet);
+            for (let resultSetId = 0; resultSetId < batchSet.resultSetSummaries.length; resultSetId++) {
```

Comment:
So for every batch you send out a batchStart followed immediately by a batchComplete? Is that just to have the new tab "catch up" with the completed results? What happens if a batch is still executing when it's switched?

Reply:
https://github.com/microsoft/vscode-mssql/pull/1282/files#diff-460f7bbdef98ad458411a73a20676d65R57